### PR TITLE
Unit test on "-" string values in yaml maps

### PR DESCRIPTION
**Pull request by olivielpeau**
### What does this PR do?

* Test reproducing a bug on "-" string values in yaml maps in the `secrets` pkg
* Fix to the config template docs of `tag_value_split_separator`.

The test fails without a revert of the PR that caused the bug (https://github.com/DataDog/datadog-agent/pull/6593). After merging master (which include the revert) into this branch, the test correctly passes.

### Motivation

Add a regression test and fix docs.

### Additional Notes

Supersedes https://github.com/DataDog/datadog-agent/pull/6585

### Describe your test plan

N/A

### CODE REVIEW

File: pkg/secrets/secrets_test.go [hunk 1]
```diff
@@ -188,6 +199,32 @@ func TestDecryptSecretError(t *testing.T) {
...
+
+	defer func() {
+		secretBackendCommand = ""
+		secretCache = map[string]string{}
+		secretOrigin = map[string]common.StringSet{}
+		secretFetcher = fetchSecret
+	}()
+
+	secretFetcher = func(secrets []string, origin string) (map[string]string, error) {
+		sort.Strings(secrets)
```
Comment (ogaca-dd) [inline pkg/secrets/secrets_test.go:213]: Not required.

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: pkg/secrets/secrets_test.go (hunk 1 above)
Comment (olivielpeau) [reply pkg/secrets/secrets_test.go]: good catch, fixed

**Comment by olivielpeau** [comment]
Confirmed that the test fails (commit 96a3a91) without the needed revert, and correctly succeeds once master (which includes the revert) is merged into this branch (commit 878ba9d).

Merging

**Pull request merged**